channeldb/invoices.go

```diff
@@ -1381,7 +1381,10 @@ func (d *DB) updateInvoice(hash lntypes.Hash, invoices, settleIndex *bbolt.Bucke
 		return &invoice, nil
 	}
 
-	now := d.Now()
+	// Strip the monotonic clock reading. This is helpful when
+	// deep comparing time points in tests where updateInvoice
+	// has updated timestamps and expectations are pre generated.
+	now := d.Now().Round(0)
```

Comment:
Looking at this again, my opinion is that we shouldn't need this. For testing, we should to inject a mocked `Now()` to make the test deterministic and also not have the problem with the monotonic clock.

Reply:
After some offline conversation, decided to use fixed Now() in the tests instead.